core/src/main/java/com/xatkit/core/recognition/nlpjs/NlpjsIntentRecognitionProvider.java

```diff
@@ -0,0 +1,210 @@
+                    EntityDefinition.class.getSimpleName()));
+        }
+    }
+
+    @Override
+    public void registerIntentDefinition(@NonNull IntentDefinition intentDefinition) throws IntentRecognitionProviderException {
+        checkNotShutdown();
+        checkNotNull(intentDefinition.getName(), "Cannot register the %s with the provided name %s",
+                IntentDefinition.class.getSimpleName(), intentDefinition.getName());
+        if (this.intentsToRegister.containsKey(intentDefinition.getName())) {
+            throw new IntentRecognitionProviderException(MessageFormat.format("Cannot register the intent {0}, the " +
+                    "intent already exists", intentDefinition.getName()));
+        }
+        Log.debug("Registering NLP.js intent {0}", intentDefinition.getName());
+        Intent intent = nlpjsIntentMapper.mapIntentDefinition(intentDefinition);
+        this.intentsToRegister.put(intentDefinition.getName(), intent);
+    }
+
+    @Override
+    public void deleteEntityDefinition(@NonNull EntityDefinition entityDefinition) throws IntentRecognitionProviderException {
+
+    }
+
+    @Override
+    public void deleteIntentDefinition(@NonNull IntentDefinition intentDefinition) throws IntentRecognitionProviderException {
+
+    }
+
+    @Override
+    public void trainMLEngine() throws IntentRecognitionProviderException {
+        checkNotShutdown();
+        Log.info("Starting NLP.js agent training (this may take a few minutes)");
+        TrainingData trainingData = TrainingData.newBuilder()
+                .config(new AgentConfig(this.configuration.getLanguageCode()))
+                .intents(new ArrayList<>(this.intentsToRegister.values()))
+                .entities(new ArrayList<>(this.entitiesToRegister.values()))
+                .build();
+        try {
+            boolean isDone = false;
+            int attemptsLeft = 10;
+            this.nlpjsService.trainAgent(agentId, trainingData);
+            while (!isDone && attemptsLeft > 0) {
+                Thread.sleep(2000);
+                Agent agent = this.nlpjsService.getAgentInfo(this.agentId);
+                if (agent.getStatus().equals(AgentStatus.READY))
+                    isDone = true;
+                else
+                    attemptsLeft--;
+            }
+            if (isDone)
+                Log.info("NLP.js agent trained.");
+            else
+                throw new IntentRecognitionProviderException("Failed to train the NLP.js agent");
+        } catch (IOException | InterruptedException e) {
+            throw new IntentRecognitionProviderException("An error occurred during the NLP agent training: ", e);
+        }
+    }
+
+    @Override
+    public XatkitSession createContext(@NonNull String sessionId) throws IntentRecognitionProviderException {
+        return new XatkitSession(sessionId);
+    }
+
+    @Override
+    public void shutdown() throws IntentRecognitionProviderException {
+
```

Comment:
Are you sure there is nothing to close/shutdown here?

Reply:
The RecognitionMonitor should be shutdown (see DialogFlowApi's implementation)

Reply:
I'll check